db/table_cache.cc

```diff
@@ -125,10 +126,19 @@ Status TableCache::GetTableReader(
       file->Hint(FSRandomAccessFile::kRandom);
     }
     StopWatch sw(ioptions_.clock, ioptions_.stats, TABLE_OPEN_IO_MICROS);
+    IOActivity io_activity;
+    if (caller == TableReaderCaller::kFlush) {
+      io_activity = IOActivity::kFlush;
+    } else if (caller == TableReaderCaller::kCompaction ||
+               caller == TableReaderCaller::kCompactionRefill) {
+      io_activity = IOActivity::kCompaction;
+    } else {
+      io_activity = IOActivity::kUnknown;
+    }
     std::unique_ptr<RandomAccessFileReader> file_reader(
         new RandomAccessFileReader(
             std::move(file), fname, ioptions_.clock, io_tracer_,
-            record_read_stats ? ioptions_.stats : nullptr, SST_READ_MICROS,
+            record_read_stats ? ioptions_.stats : nullptr, io_activity,
```

Comment:
This `io_activity` probably makes sense to pass to `NewTableReader()` and use it to attribute prefetches/reads that happen there

Reply:
Fixed